localstripe/resources.py

```diff
@@ -1400,6 +1395,70 @@ def _api_list_all(cls, url, customer=None, limit=None):
         return li
 
 
+class InvoiceLineItem(StripeObject):
+    object = 'line_item'
+    _id_prefix = 'il_'
+
+    def __init__(self, item):
+        try:
+            assert isinstance(item, (InvoiceItem, SubscriptionItem))
+        except AssertionError:
+            raise UserError(400, 'Bad request')
+
+        # All exceptions must be raised before this point.
+        super().__init__()
+
+        self.type = \
+            'invoiceitem' if isinstance(item, InvoiceItem) else 'subscription'
+
+        if self.type == 'subscription':
+            self.subscription_item = item.id
+            self.subscription = item._subscription
+        elif self.type == 'invoiceitem':
+            self.invoice_item = item.id
+            self.subscription = None
+
+        # Legacy support, before InvoiceLineItem
+        self.invoice = item.invoice
+
+        self.plan = item.plan
+        self.tax_rates = item.tax_rates or []
+        self.metadata = item.metadata
+
+        self.amount = item.amount \
+            if self.type == 'invoiceitem' else item._calculate_amount()
+        self.currency = \
+            item.currency if self.type == 'invoiceitem' else item.plan.currency
+        self.description = \
+            item.description if self.type == 'invoiceitem' else item.plan.name
+        self.proration = \
+            item.proration if self.type == 'invoiceitem' else False
```

Comment:
`proration` can only exist when the `InvoiceLineItem` references an `InvoiceItem`.
Using `False` as the default value here for `SubscriptionItem`s.